### PR TITLE
Allow teams page navigation from team on mobile

### LINKED ISSUE #1925: Cannot navigate to base teams page from a team on mobile
On mobile, we disable the currently selected page in the header hamburger menu. Need to remove that logic, at least for teams

### PR DESCRIPTION
**Pull request by frzyc**
## Issue or discord link

- Resolves #1925 

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/frontend/src/app/Header.tsx [hunk 1]
```diff
@@ -334,7 +336,7 @@ function MobileHeader({
                   component={RouterLink}
                   to={to}
                   selected={currentTab === value}
-                  disabled={currentTab === value}
+                  disabled={currentTab === value && !inTeam}
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Header.tsx:339]: Why not allow navigating back all the time? We allow it in desktop view, I don't really see a reason to disable it

Comment (frzyc) [reply apps/frontend/src/app/Header.tsx:339]: It acts as more of an indicator to which page the user is on. we could probably style the buttons differently to indicate, but this is the fastest solution

Comment (nguyentvan7) [reply apps/frontend/src/app/Header.tsx:339]: Could make it blue like desktop

Comment (frzyc) [reply apps/frontend/src/app/Header.tsx:339]: could, if you feel strongly about it, raise an issue. this PR is mainly address the bug issue #1925.

Comment (nguyentvan7) [reply apps/frontend/src/app/Header.tsx:339]: its like 3 line change, and it'll have to undo this

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**